src/components/toggle/toggle.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {SBComponent} from 'stylable-react-component';
+import style from './toggle.st.css';
+
+export interface Props {
+    checked?: boolean,
+    error?: boolean,
+    disabled?: boolean,
+    onChange?: (selected: boolean) => void,
+    label?: string,
+    tabIndex?: number,
+    rtl?: boolean
+}
+export interface State {
+    focus: boolean,
+    click: boolean
+}
+
+@SBComponent(style)
+export default class Toggle extends React.Component<Props, State> {
+    static defaultProps = {
+        checked: false,
+        disabled: false,
+        error: false,
+        rtl: false
+    }
+    state = {
+        focus: false,
+        click: false
+    }
+    toggle = (e: React.SyntheticEvent<HTMLInputElement>) => {
+        if (!this.props.disabled && this.props.onChange) {
+            this.props.onChange(!this.props.checked)
+        }
+        // reseting focus when user click toggle
```

Comment:
Still confused about this despite the comment. Why do we want to reset the focus? I can't find anything about this in the spec.